loc-analysis.exs

```diff
@@ -0,0 +1,202 @@
+#! /usr/bin/env elixir
+
+Mix.install([
+  {:csv, "~> 3.0"}
+])
+
+defmodule GitCommandError do
+  defexception message: "Git command failed."
+
+  @impl true
+  def exception(value) do
+    case value do
+      [] -> %GitCommandError{}
+      _ -> %GitCommandError{message: "Git command failed: " <> value}
+    end
+  end
+end
+
+defmodule Git do
+  def get_snapshot!(commit_hash) do
```

Comment:
Can you elaborate on what you mean by "snapshot"? Is it a snapshot in the git terminology, or is something else? (I'm aware of a snapshot in git as being a state of the working tree, but this function seems to only return a path-like string concatenating an actual path with a commit hash. Maybe that's the same thing but I have troubles parsing that.)

Reply:
> snapshot in git as being a state of the working tree

Yes exactly. It captures the repo state at the given commit into a temp directory then returns the path of the directory.

Maybe I could name it more clearly. Or should I break this function since it's doing too much?

Reply:
I don't really believe it's doing too much, but maybe the the pipe you use in the fourth expression within the `with` could be itself a function (I mean `Path.join(destination_path, tar_file) |> File.rm!()`) -- `with` can easily become very dense, so I like when expressions on the rhs of `<-` are very simple calls. It may be totally personal taste!